DomainContent/Halloween/flashlightSpawner.js

```diff
@@ -0,0 +1,174 @@
+//
+//  FlashlightSpawner.js
+//
+//  created by Rebecca Stankus on 09/19/18
+//  Copyright 2018 High Fidelity, Inc.
+//
+//  Distributed under the Apache License, Version 2.0.
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+//
+
+/* global Pointers */
+
+(function() { 
+    var _this;
+
+    var SEARCH_RADIUS = 0.1;
+    var CHECKING_INTERVAL_MS = 5000;
+
+    var checking;
+
+    function FlashlightSpawner() {
+        _this = this;
+    }
+
+    FlashlightSpawner.prototype = {
+        position: null,
+        preload: function(entityID) {
+            _this.entityID = entityID;
+            _this.position = Entities.getEntityProperties(_this.entityID, 'position').position;
+            _this.checkingStatus();
+        },
+
+        checkingStatus: function() {
+            checking = Script.setInterval(function() {
+                var needClone = true;
+                Entities.findEntities(_this.position, SEARCH_RADIUS).forEach(function(element) {
+                    var name = Entities.getEntityProperties(element, 'name').name;
+                    if (name === "Flashlight") {
+                        print("found another flashlight ", element, " at ", JSON.stringify(
+                            Entities.getEntityProperties(element, 'position').position));
+                        needClone = false;
+                    }
+                });
+                if (needClone) {
+                    _this.spawnNewFlashlight();
+                }
+            }, CHECKING_INTERVAL_MS);
+        },
+
+        spawnNewFlashlight: function() {
+
+            var parent = Entities.addEntity({
+                clientOnly: false,
+                collidesWith: "static,dynamic,kinematic,",
+                collisionMask: 7,
+                lifetime: 1800,
+                dimensions: {
+                    x: 0.374295175075531,
+                    y: 0.2721865177154541,
+                    z: 0.24757488071918488
+                },
+                visible: true,
+                modelURL: "https://hifi-content.s3.amazonaws.com/jimi/environment/201804_tomb/BuildersTorch.fbx?1?1",
+                name: "Flashlight",
+                serverScripts: Script.resolvePath("emptyServerScript.js"),
```

Comment:
Should this be in the PR next to this script?

Reply:
oh, yes